src/client/client.ts

```diff
@@ -203,6 +379,74 @@ export default class CTProtoClient<AuthRequestPayload, AuthResponsePayload, ApiR
     });
   }
 
+  /**
+   * Handle chunk response from server
+   *
+   * @param fileId - uploading file id
+   * @param chunkNumber - file chunk number, which handled on server
+   */
+  private onChunkResponseMessage( fileId: string, chunkNumber: number ): void {
+    const uploadingFile = this.getUploadingFileById( fileId );
+
+    if (!uploadingFile) {
+      return;
+    }
+
+    /**
+     * Make log about loading progress
+     */
+
+    const percentMultiplier = 100;
+    const percent = Math.floor(chunkNumber / Math.ceil(uploadingFile.chunks.length / this.bufferLimit) * percentMultiplier);
+
+    this.log('File ' + fileId + ' uploaded on: ' + percent + '%');
+
+    /**
+     * Implement sending chunk, because previous was uploaded
+     */
+    uploadingFile.sendingChunk = uploadingFile.sendingChunk + 1;
```

Comment:
looks strange bytheway. Why the file contains the number of uploaded chunks? They could be uploaded in a random order.